azext_iot/iothub/_help.py

```diff
@@ -309,3 +309,103 @@ def load_iothub_help():
         type: command
         short-summary: Upload a local file as a device to a pre-configured blob storage container.
     """
+
+    helps[
+        "iot hub state"
+    ] = """
+        type: group
+        short-summary: Manage the state of an IoT Hub.
+    """
+
+    helps[
+        "iot hub state export"
+    ] = """
+        type: command
+        short-summary: Export the state of an IoT Hub to a file.
+        long-summary: |
+                       The exported state will include: arm template for hub, hub configurations (including ADM
+                       configurations and edge deployments), device information (including device identites,
+                       device twins, module identities and module twins).
+        examples:
+        - name: Export the supported state of the specified hub to the specified file.
+          text: >
+            az iot hub state export -n {iothub_name} -f {state_filename}
+        - name: Export the supported state of the specified hub to the specified file, overwriting the file contents.
+          text: >
+            az iot hub state export -n {iothub_name} -f {state_filename} -r
+        - name: Export only the devices and configurations of the specified hub to the specified file.
+          text: >
+            az iot hub state export -n {iothub_name} -f {state_filename} --aspects devices configurations
+    """
+
+    helps[
+        "iot hub state import"
+    ] = """
+        type: command
+        short-summary: Import a Hub state from a file to an IoT Hub.
+        long-summary: |
+                       If the arm aspect is specified, the hub will be created if it does not exist.
+
+                       The imported state will include: arm template for hub, hub configurations (including ADM
+                       configurations and edge deployments), device information (including device identites,
+                       device twins, module identities and module twins).
+
+                       For imported endpoints with system assigned identity authentication, the specified hub must have
```

Comment:
In the case a new IoT Hub is provisioned, what is the expectation in how the user would set system assigned identity role assignments?

Reply:
Right now, the command will just turn on the system assigned identity but not assign roles - so having a system assigned identity endpoint would fail. I added some instructions on how we expect customers to deal with this:

To fix this, add the permissions needed for the system assigned identity and rerun the command. If this cannot be done because the hub does not exist yet (or system assigned permissions are not turned on):
 1. make a copy of the state you want to import and remove the offending endpoint(s) in your copy
 2. run `az iot hub state import` with the copied file
 3. check that the hub has been created and system assigned identity is turned on
 4. assign the correct permissions for the IoT Hub's system assigned identity to the correct endpoint scope(s)
 5. run `az iot hub state import` with the original file (with the endpoint(s)) or create the endpoint manually